src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,178 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int ZERO = 0;
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(value);
+        if (first == null) {
+            node.next = null;
+            node.prev = null;
+            first = node;
+        } else {
+            last.next = node;
+            node.prev = last;
+        }
+        last = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index " + index + " absent in this list");
+        }
+        Node<T> node = new Node<>(value);
+        if (index == 0) {
+            if (first == null) {
+                node.next = null;
+                node.prev = null;
+                first = node;
+                last = node;
+            } else {
+                first.prev = node;
+                node.next = first;
+                first = node;
+                node.prev = null;
+            }
+            size++;
+            return;
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> tempNode = first;
+        for (int i = 0; i < index; i++) {
+            tempNode = tempNode.next;
+        }
+        Node<T> previous = tempNode.prev;
+        previous.next = node;
+        tempNode.prev = node;
+        node.prev = previous;
+        node.next = tempNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + " absent in this list");
+        }
+        Node<T> tempNode = first;
+        for (int i = 0; i < index; i++) {
+            tempNode = tempNode.next;
+        }
+        return tempNode.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + " absent in this list");
+        }
+        Node<T> node = new Node<>(value);
+        T returnedValue;
+        if (index == 0) {
+            node.prev = null;
+            node.next = first.next;
+            returnedValue = first.item;
+            first = node;
+            return returnedValue;
+        }
+        Node<T> tempNode = first;
+        for (int i = 0; i < index; i++) {
+            tempNode = tempNode.next;
+        }
```

Comment:
we can create one more separete method to get Node by Index and use it in diferent cases